test/cmd/admin.sh

```diff
@@ -41,188 +42,192 @@ status:
 # Ensure --additive-only=false removes customized users from the binding
-oadm policy reconcile-cluster-role-bindings --additive-only=false --confirm
-[ ! "$(oc get clusterrolebindings/basic-users -o json | grep custom-user)" ]
+os::cmd::expect_success 'oadm policy reconcile-cluster-role-bindings --additive-only=false --confirm'
+os::cmd::expect_success_and_not_text 'oc get clusterrolebindings/basic-users -o json' 'custom-user'
 
 echo "admin-policy: ok"
 
 # Test the commands the UI projects page tells users to run
 # These should match what is described in projects.html
-oadm new-project ui-test-project --admin="createuser"
-oadm policy add-role-to-user admin adduser -n ui-test-project
+os::cmd::expect_success 'oadm new-project ui-test-project --admin="createuser"'
+os::cmd::expect_success 'oadm policy add-role-to-user admin adduser -n ui-test-project'
 # Make sure project can be listed by oc (after auth cache syncs)
-tryuntil '[ "$(oc get projects | grep "ui-test-project")" ]'
+tryuntil "os::cmd::expect_success_and_text 'oc get projects' 'ui-test-project'"
 # Make sure users got added
-[ "$(oc describe policybinding ':default' -n ui-test-project | grep createuser)" ]
-[ "$(oc describe policybinding ':default' -n ui-test-project | grep adduser)" ]
+os::cmd::expect_success_and_text 'oc describe policybinding ':default' -n ui-test-project' 'createuser'
+os::cmd::expect_success_and_text 'oc describe policybinding ':default' -n ui-test-project' 'adduser'
 echo "ui-project-commands: ok"
 
 
 # Test deleting and recreating a project
-oadm new-project recreated-project --admin="createuser1"
-oc delete project recreated-project
-tryuntil '! oc get project recreated-project'
-oadm new-project recreated-project --admin="createuser2"
+os::cmd::expect_success 'oadm new-project recreated-project --admin="createuser1"'
+os::cmd::expect_success 'oc delete project recreated-project'
+tryuntil "os::cmd::expect_failure 'oc get project recreated-project'"
+os::cmd::expect_success 'oadm new-project recreated-project --admin="createuser2"'
 oc describe policybinding ':default' -n recreated-project | grep createuser2
 echo "new-project: ok"
 
 # Test running a router
-[ ! "$(oadm router --dry-run | grep 'does not exist')" ]
-echo '{"kind":"ServiceAccount","apiVersion":"v1","metadata":{"name":"router"}}' | oc create -f - -n default
-oc get scc privileged -o yaml | sed '/users:/ a\
+os::cmd::expect_failure_and_text 'oadm router --dry-run' 'does not exist'
+encoded_json='{"kind":"ServiceAccount","apiVersion":"v1","metadata":{"name":"router"}}'
+os::cmd::expect_success "echo '${encoded_json}' | oc create -f - -n default"
+os::cmd::expect_success "oc get scc privileged -o yaml | sed '/users:/ a\
 - system:serviceaccount:default:router\
-' | oc replace scc privileged -f -
-[ "$(oadm router -o yaml --credentials="${KUBECONFIG}" --service-account=router -n default | egrep 'image:.*-haproxy-router:')" ]
```

Comment:
I got rid of `egrep`, which is deprecated, and redundant in this case as we run `grep -Eq` in the wrapper function.